### DISCUSSION TITLE
Installation on Synology Docker

### DISCUSSION THREAD

**Discussion by CommittotheIndian** [briis/hass-weatherflow2mqtt#42] [Q&A]
I've been trying to install this image on my Synology Docker (HA is on a Blue). I have docker and an MQTT broker on Synology and a MQTT broker on the Blue.

I've tried multiple variations of the docker setup (installed via command line & via portioner). I keep getting similar errors.

`INFO:__main__:Timezone is America/Indiana
ERROR:helpers:Could not find VERSION File.
INFO:__main__:Connected to the MQTT server on address 10.10.10.10 and port 1883...
ERROR:__main__:Could not start listening to the UDP Socket. Error is: [Errno 99] Cannot assign requested address`

If I install via command line, I seem to get the "Could not find VERSION File" under all lines. At least portioner I just get the one error.

I also tried to install via some of the instruction for the Run on ARM Architecture (https://github.com/briis/hass-weatherflow2mqtt/discussions/10). Mostly the Container deployment. When I do that, I can't edit the ENV settings.

Any suggestions?

**Comment by briis** [comment]
No immediate suggestions. Let me test it out on my own synology and then come back to you.

**Reply by CommittotheIndian** [reply]
Wondering if you've come up with anything for this?

**Reply by CommittotheIndian** [reply]
I tried reinstalling again. Tried some changes with the IP for the device. So far it is continuing to run (at about 5 mins now) but still getting an error. This is the whole log after 5 mins:

`DEBUG:asyncio:Using selector: EpollSelector
INFO:__main__:Timezone is America/Indiana/Indianapolis
ERROR:helpers:Could not find VERSION File.
INFO:__main__:Connected to the MQTT server on address 10.10.10.10 and port 1883...
INFO:__main__:The UDP server is running on port 50222...
INFO:__main__:Defining Sensors for Home Assistant`

I'm not sure what to do about VERSION file. Nothing is showing up in HA yet either. The Tempest is on the 10.10.10.* network with the MQTT but the docker is on a 10.1.1.* network. There is an MQTT bridge between the two networks though so I know they communicate. I've also been running MQTT Explorer, but not seeing anything for the Tempest.

**Reply by CommittotheIndian** [reply]
I've tried yet another install but this time used the mqtt broker on the same device as my container. Still the same errors. I don't have an SQLite Database anywhere.

**Reply by briis** [reply]
I am tight up with work right now, but hopefully I can do some testing later this week.

**Comment by chasut** [comment]
I run in docker on the Synology.  I just use the basic synology docker container manager and the Synology Mosquitto package for MQTT.  Pretty simple and works great.  Happy to help if I can.  The only real challenge I had was to figure out how to get pull the image from GitHub, as it doesn't show up in the standard registries in the Synology interface.  Ended up doing the docker pull from the command line "docker pull ghcr.io/briis/hass-weatherflow2mqtt:latest".  Now that I had the image, it was just a bit of translating the command line run command that Bjarne has in the repo into the synology docker interface.  While that interface is overly simplified in my view, I do like the fact that is easy to add/change the environment variables.

I did have a somewhat similar error initially with the port mapping.  I don't remember the specifics, but I ended up just mapping 50222 udp and not the MQTT port 1883.  The container was happy and everything works great.

Let me know if you have questions about my config.

**Comment by chasut** [comment]
Oops, just saw this after my reply...  "The Tempest is on the 10.10.10.* network with the MQTT but the docker is on a 10.1.1.* network."  

FYI, I fought with trying to route the UDP traffic across VLANS and was not very successful.  Ended up just putting the tempest on the same network as the container until I figure out a more elegant solution.  The MQTT traffic was not a problem.

If you figure out how to reliably route the UDP traffic, I am all ears :)

**Reply by CommittotheIndian** [reply]
Thanks for reaching out. 

I’ll try again when I get home. 
My tempest is on 10.10.10
I have MQTT on 10.1.1. & 10.10.10. I’ve tried both and mostly recently the same as tempest. 
My Synology is on 10.1.1. 

So the rocker running everything would be on 10.1.1 but tempest and mqtt would be 10.10.10. 

I could either change tempest to 10.1.1 or more preferably, any ideas how to install a rocker container on a Home Assistant (o.droid)? I could then try installing the docker on the same network.
